pkgs/tools/misc/edk2-uefi-shell/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib
+, edk2
+, util-linux
+, nasm
+, python3
+, ...
+}:
+(edk2.mkDerivation "ShellPkg/ShellPkg.dsc" {
+  pname = "edk2-uefi-shell";
+  inherit (edk2) version;
+
+  nativeBuildInputs = [ util-linux nasm python3 ];
+  strictDeps = true;
+
+  # We only have a .efi file in $out which shouldn't be patched or stripped
+  dontPatchELF = true;
+  dontStrip = true;
+
+  # GUID hardcoded to match ShellPkg.dsc
+  installPhase = ''
+    runHook preInstall
+    install -D -m0644 Build/Shell/RELEASE*/*/Shell_EA4BB293-2D7F-4456-A681-1F22F42CD0BC.efi $out/shell.efi
+    runHook postInstall
+  '';
+
+  meta = {
+    inherit (edk2.meta) license platforms;
+    description = "UEFI Shell from Tianocore EFI development kit";
+    homepage = "https://github.com/tianocore/tianocore.github.io/wiki/ShellPkg";
+    maintainers = with lib.maintainers; [ LunNova ];
+  };
+}).overrideAttrs (finalAttrs: previousAttrs: {
+  passthru.efi = "${finalAttrs.finalPackage}/shell.efi";
```

Comment:
Please inline.

Reply:
Doesn't work currently, unless I also swap it to use placeholder out as edk2's mkDerivation expects an attrset, so I'll do that for now.

Reply:
Thats totally fine to do.

Reply:
Done with placeholder for now, if I add `finalPackage` support to `edk2.mkDerivation` it'll be in a future PR.